### ISSUE TITLE
Results not reproducible

### ISSUE THREAD

**Issue by fjulian** [EmilienDupont/wgan-gp#4]
Thanks for sharing this repo.

Running the code (`main.py`, 200 epochs, parameters as mentioned in readme, only fixing the the issue mentioned in #1), I couldn't reproduce the results shown in https://github.com/EmilienDupont/wgan-gp/blob/master/gifs/mnist_200_epochs.gif. Instead, the generated samples look as follows:

![training_200_epochs](https://user-images.githubusercontent.com/10942150/145377412-4f90c725-a286-4eb3-9195-b0b282fa180e.gif)

Any hints on what would need to be changed to make the training successful?

One step towards reproducibility would be to fix random seeds in the beginning of `main.py`, e.g. with the following code:

```python
import random
import numpy as np
import torch
random_seed = 42
random.seed(random_seed)
np.random.seed(random_seed)
torch.manual_seed(random_seed)
torch.cuda.manual_seed_all(random_seed)
```

**Comment by echocxl** [comment]
Hello ,I met the same problem as you.
And I find it that this gif what my model generated 6.60Mb,
But in this code example gif folder,it is 8.22 Mb.
What make this difference?
I hope author to anser this question,thank you very much!

**Comment by zhmou** [comment]
> Hello ,I met the same problem as you. And I find it that this gif what my model generated 6.60Mb, But in this code example gif folder,it is 8.22 Mb. What make this difference? I hope author to anser this question,thank you very much!

guess it's the learning rate and trainning epoch causing that problem, i changed it to 5x larger: lr = 5e-4, and the performance will be better:
after 50 epochs trainning:
![Epoch50](https://user-images.githubusercontent.com/43105172/162710510-12928b92-eb09-4049-a681-c004e8cd48ca.png)

(you can train it longer and check your result)
